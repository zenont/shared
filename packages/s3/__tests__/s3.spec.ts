/* eslint-disable @typescript-eslint/no-explicit-any */
import createS3 from '../src'

const mockedGetObject = jest.fn()
const mockedPutObject = jest.fn()
jest.mock('aws-sdk', () => ({
  __esModule: true,
  default: {
    S3: class S3 {
      putObject(params: any, cb: any) {
        mockedPutObject(params, cb)
        cb(undefined, {
          VersionId: `Fake VersionId: ${JSON.stringify(params.Body)}`
        })
      }
      getObject(params: any, cb: any) {
        mockedGetObject(params, cb)
        cb(undefined, { Body: JSON.stringify(params.Key) })
      }
    }
  }
}))

describe('s3', () => {
  it('should get JSON', async () => {
    const s3 = createS3()
    const result = await s3.bucket('holos-s3').getJSON('my-test')
    expect(result).toMatchSnapshot()
  })

  it('should put JSON', async () => {
    const s3 = createS3()
    const result = await s3.bucket('holos-s3').putJSON('my-test', { test: 1 })
    expect(result).toMatchSnapshot()
  })

  it('should get base64', async () => {
    const s3 = createS3()
    const result = await s3.bucket('holos-s3').getBase64('ImVuY29kZW1lIg==')
    expect(result).toMatchSnapshot()
  })

  it('should put base64', async () => {
    const s3 = createS3()
    const result = await s3.bucket('holos-s3').putBase64('my-test', 'encodeme')
    expect(result).toMatchSnapshot()
  })
})
