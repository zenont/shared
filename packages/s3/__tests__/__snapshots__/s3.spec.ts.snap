// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`s3 should get JSON 1`] = `"my-test"`;

exports[`s3 should get base64 1`] = `"\\"encodeme\\""`;

exports[`s3 should put JSON 1`] = `"Fake VersionId: \\"{\\\\\\"test\\\\\\":1}\\""`;

exports[`s3 should put base64 1`] = `"Fake VersionId: \\"ImVuY29kZW1lIg==\\""`;
